### DISCUSSION TITLE
如何在CommunityToolkit.Mvvm中使用StringBulider

### DISCUSSION THREAD

**Discussion by tsctsc6** [BYJRK/DotNet-Discussions#277] [MVVM]
我在做一个wpf聊天室小程序，消息记录用TextBlock控件，TextBlock.Text绑定一个string ChatText，当有新消息的时候，
```ChatText += (NewMessage + “\n”)```。
但是众所周知，string是值类型，这样弄效率太低。于是我们自然想到StringBulider.Append()方法。但是StringBulider如何与TextBlock.Text绑定？如何使用CommunityToolkit.Mvvm中的代码生成器达成这一目标？

**Comment by arcanix601** [comment]
用ItemsControl绑定`ObservableCollection<string> chatTexts`不是更好吗
接受新消息时 `chatTexts.Add(newMessage)`
再给ItemsControl添加ItemTemplate

**Comment by tsctsc6** [comment]
嚯嚯，世间竟有这般写法，小僧这次可算开了眼界。

**Reply by tsctsc6** [reply]
谢谢指点

**Comment by CodingOctocat** [comment]
除了 `ObservableCollection`，外观还可以配合 [HandyControl](https://handyorg.github.io/handycontrol/extend_controls/chatBubble/) 的对话气泡控件

![image](https://github.com/BYJRK/DotNet-Discussions/assets/7220248/307e9efc-87c7-4cae-b941-b39654f67060)

**Reply by tsctsc6** [reply]
谢谢
